src/main/java/de/tum/in/www1/artemis/web/rest/CompetencyResource.java

```diff
@@ -260,7 +275,41 @@ public ResponseEntity<Competency> importCompetency(@PathVariable long courseId,
     }
 
     /**
-     * Imports all competencies of the source course (and optionally their relations) into another
+     * POST /courses/:courseId/competencies/import/bulk : imports a number of competencies (and optionally their relations) into a course.
+     *
+     * @param courseId             the id of the course to which the competencies should be imported to
+     * @param competenciesToImport the competencies that should be imported
+     * @param importRelations      if relations should be imported aswell
+     * @return the ResponseEntity with status 201 (Created) and with body containing the imported competencies
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("/courses/{courseId}/competencies/import/bulk")
+    @EnforceAtLeastEditor
+    public ResponseEntity<List<CompetencyWithTailRelationDTO>> importCompetencies(@PathVariable long courseId, @RequestBody List<Competency> competenciesToImport,
+            @RequestParam(defaultValue = "false") boolean importRelations) throws URISyntaxException {
+        log.info("REST request to import competencies: {}", competenciesToImport);
+
+        var course = courseRepository.findWithEagerCompetenciesByIdElseThrow(courseId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.EDITOR, course, null);
+
+        var competencies = new HashSet<>(competenciesToImport);
+        List<CompetencyWithTailRelationDTO> importedCompetencies;
+
+        if (importRelations) {
```

Comment:
Does this belong to the resource or should this rather be part of a service?

Reply:
In my opinion the logic is simple enough to be in the Resource. Otherwise I would need to add two more methods to the service (`importCompetenciesBulk` and `importAllCompetenciesFromCourse`), which makes the code more complex and which might lead to confusion since they call similar named methods `importCompetencies`/`importCompetenciesAndRelations`